### PR TITLE
Adding service account to proxy postsubmit test

### PR DESCRIPTION
**Pull request by utka**

### CODE REVIEW

**Comment by cjwagner** [comment]
`service-account` is the secret we use for uploading job results to `gs://istio-prow`. I know that this secret also has permission to upload to `gs://istio-build`, but that should be revoked. Ideally we should be using a separate service account for upload to `gs://istio-build` instead of sticking all permissions that might be needed into one service account.
This service account will work though...
/shrug

I think you'll also need to specify the `GOOGLE_APPLICATION_CREDENTIALS` env var. e.g. https://github.com/kubernetes/test-infra/blob/5de6cdf6fe7bb5277afdb0884afd8356154820a3/config/jobs/kubernetes/test-infra/test-infra-trusted.yaml#L50-L59

**Comment by utka** [comment]
GOOGLE_APPLICATION_CREDENTIALS already specified.

File: prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml [hunk 1]
```diff
@@ -124,5 +124,9 @@ postsubmits:
           limits:
             memory: "60Gi"
             cpu: "16000m"
+      volumes:
```
Comment (howardjohn) [inline prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml:127]: do we need a volumeMount?

Comment (howardjohn) [reply prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml:127]: Also doesn't `preset-service-account: "true"` do this already?

Comment (utka) [reply prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml:127]: I am trying to figure out how to make the test be happy :|

Comment (utka) [reply prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml:127]: https://github.com/istio/istio/issues/15714#issuecomment-517483833

Comment (howardjohn) [reply prow/cluster/jobs/istio/proxy/istio.proxy.master.yaml:127]: We have plenty of other jobs that push to GCR though.. every e2e test in istio/istio does and just ` preset-service-account: "true"` is sufficient. The ` preset-service-account: "true"` preset looks like it already adds this volume actually?

**Comment by utka** [comment]
Not necessary

**Pull request closed without merging**